src/pages/workspaces/workspace/analysis/runtime-utils.ts

```diff
@@ -0,0 +1,142 @@
+import _ from 'lodash/fp'
```

Comment:
Most changes in this PR stem from updating this file to typescript.

This was done to validate the Runtime model I created, as most of these methods are data transformation on `Runtimes`, and is decently tested already. A good bit of code unrelated to `Runtime` models werre also moved  to a better home.